app/controllers/registrant/contacts_controller.rb

```diff
@@ -26,4 +27,8 @@ def domain_ids
       BusinessRegistryCache.fetch_by_ident_and_cc(ident, ident_cc).associated_domain_ids
     end
   end
+
+  def domain
+    Domain.find(params[:domain_id])
```

Comment:
Not necessarily true https://guides.rubyonrails.org/caching_with_rails.html#sql-caching.

In general, I agree that `||=` is better, but I think it is still a premature optimization, and it's not worth paying too much attention to this.

Reply:
There's an additional problem with that, at the moment it's visible only in the breadcrumb, but it might baloon to something easier.

1. Go to any registrant's contact show page.
2. In URL, in id field enter an id from an existing domain that does not belong to the same Registrant.
3. Observe the breadcrumb - it will have a domain name from a domain that does not belong to the same user.

![image](https://user-images.githubusercontent.com/2331765/44093207-8790a960-9fdb-11e8-93e7-a3d15187f718.png)

Reply:
Good catch! Not sure if it's worth the effort, since I had to copy-paste some some that needs refactoring.

Anyway, it is now fixed. The bug is quite nasty, that's for sure.